google/cloud/storage/_helpers.py

```diff
@@ -449,6 +466,29 @@ def _convert_to_timestamp(value):
     return mtime
 
 
+def _add_etag_match_headers(headers, **match_parameters):
+    """Add generation match parameters into the given parameters list.
+
+    :type headers: dict
+    :param headers: Headers dict.
+
+    :type match_parameters: dict
+    :param match_parameters: if*etag*match parameters to add.
+
+    :raises: :exc:`ValueError` if ``parameters`` is not a ``dict()``.
+    """
+    if not isinstance(headers, dict):
+        raise ValueError("`headers` argument should be a dict() or a list().")
+
```

Comment:
Since this is a private helper, we can avoid this check.
```suggestion
```

Reply:
Then why does `_add_generation_match_parameters` have a check? I was just being consistent with that.

Reply:
Very good question!  The answer lies back in the mists of time, and my memory might be faulty, even with the help of `git log` and `git blame`.  `_add_generation_match_parameters` is used to add values to query parameters for a URL (versus headers here).  IIRC, in one specific case (the download URL with no `mediaLink` set), the order of parameters apparently mattered (at least, it used to) and so the function had to take *either* a `dict` or a `list`.  I don't recall why we didn't use a `collections.OrderedDict` there:  maybe we were still supporting Python 2.6 back then, and it  wasn't available.

At any rate, I'd prefer to "duck type" where possible.

Reply:
Sure, I don't particularly care about why it had to take both a list and dict, memories fade over time :D. I guess it makes sense to have a check in `_add_generation_match_parameters` because otherwise the function would just silently fail if there were no else block. `_add_etag_match_headers` will naturally raise a TypeError when trying to set a value if it isn't a dict (or dict like enough). Is that what you're getting at?

Reply:
Check removed. I also removed the raises from the docstring, even though it can still raise TypeError, because of [this from the style guide](https://google.github.io/styleguide/pyguide.html#doc-function-raises):
>You should not document exceptions that get raised if the API specified in the docstring is violated (because this would paradoxically make behavior under violation of the API part of the API).

Reply:
Yes, that is exactly what I meant.